### ISSUE TITLE
overview: Health card spacing needs adjusments

### ISSUE THREAD

**Issue by garrett** [cockpit-project/cockpit#16897]
The health card currently looks like this:

![image](https://user-images.githubusercontent.com/10246/151802284-5f51ecdf-0c81-4090-a26a-453f6ef1e758.png)

The spacing between items and within items was created in a vaccum, where just one or 2 errors would be seen at a time and none would have a link. When the user action was made, it was at the bottom and there was no other entry that would have an action below it.

Now, we have a bunch of information at once and multiple items are more than one line of text. The spacing needs to be reconsidered and adjusted accordingly so that the items are distinct from one another and would be easier to scan (that is: a reading context of "scanning"; a quick glimpse to understand).

---

As seen and mentioned in https://github.com/cockpit-project/cockpit/pull/16885#issuecomment-1023006074

**Comment by garrett** [comment]
PF spacing reference: https://www.patternfly.org/v4/guidelines/spacers/

What we need to do:

1. Adjust spacing to be a little more in between elements.
2. Adjust link spacing in the user entry to not have such a gap.
3. Additionally, the color of the icons should be considered. In this screenshot, the first two are black and the last is grey. We probably want to change that.
   - https://www.patternfly.org/v4/guidelines/colors/#typography-and-iconography-colors
   - https://www.patternfly.org/v4/guidelines/colors/#status-and-state-colors

**Comment by garrett** [comment]
This isn't exactly pixel perfect, but I did make a mockup in the GIMP by adjusting the screenshot to add a little more space between items and did tighten up the inner-item spacing. It's an example of what it should mostly look like:

![image](https://user-images.githubusercontent.com/10246/151815749-6faf2a36-a1e0-4728-9834-af3e1c58e6a9.png)

**Comment by garrett** [comment]
Icons are secondary, hence the --pf-global--Color--200 color by default (with the normal warning colors for actual issues): https://www.patternfly.org/v4/guidelines/colors/#status-and-state-colors

I wonder if we should have the bug fix icon be a different color? I guess security related ones already have the big orange ⚠ icon though, so bugfixes aren't really higher priority, else they'd be considered security updates.

Fro the font sizes: Normal text for the first (and sometimes only) line. Small for the details text, including action links. This is already the case. (No change needed.) Just restating here. :wink:

**Comment by martinpitt** [comment]
You can always do a reboot now with the reboot button on the same page, so that would feel redundant and just clutter up the card?

**Comment by garrett** [comment]
Yeah, we do have that huge reboot button. I was mainly thinking of it as an exercise of "what if there's more than one action"? But let's come to that when we need it. And we don't need it now. You're absolutely right, @martinpitt.

**Comment by jelly** [comment]
> PF spacing reference: https://www.patternfly.org/v4/guidelines/spacers/
> 
> What we need to do:
> 
> 1. Adjust spacing to be a little more in between elements.
> 2. Adjust link spacing in the user entry to not have such a gap.
> 3. Additionally, the color of the icons should be considered. In this screenshot, the first two are black and the last is grey. We probably want to change that.
>    
>    * https://www.patternfly.org/v4/guidelines/colors/#typography-and-iconography-colors
>    * https://www.patternfly.org/v4/guidelines/colors/#status-and-state-colors

The spacing is currently applied via a [css rule](https://github.com/cockpit-project/cockpit/blob/main/pkg/systemd/overview-cards/healthCard.scss#L11) should this be switched to md? So it's 16 and not 8 px.

**Comment by jelly** [comment]
> Icons are secondary, hence the --pf-global--Color--200 color by default (with the normal warning colors for actual issues): https://www.patternfly.org/v4/guidelines/colors/#status-and-state-colors
> 
> I wonder if we should have the bug fix icon be a different color? I guess security related ones already have the big orange warning icon though, so bugfixes aren't really higher priority, else they'd be considered security updates.
> 
> Fro the font sizes: Normal text for the first (and sometimes only) line. Small for the details text, including action links. This is already the case. (No change needed.) Just restating here. wink

On the standard centos-9-0
![image](https://user-images.githubusercontent.com/67428/151825609-1a66a384-42cb-4a9b-93e0-d30b9a681ce4.png)

**Comment by garrett** [comment]
> The spacing is currently applied via a css rule should this be switched to md? So it's 16 and not 8 px.

Yes, please.
